includes/class-wc-payments-token-service.php

```diff
@@ -111,6 +117,8 @@ public function woocommerce_get_customer_payment_tokens( $tokens, $user_id, $gat
 
 		$payment_methods = $this->customer_service->get_payment_methods_for_customer( $customer_id );
 
+		// Prevent unnecessary recursion, WC_Payment_Token::save() ends up calling 'woocommerce_get_customer_payment_tokens' in some cases.
+		remove_action( 'woocommerce_get_customer_payment_tokens', [ $this, 'woocommerce_get_customer_payment_tokens' ], 10, 3 );
```

Comment:
What do y'all think it's cleaner to prevent unwanted recursion? Removing the filter and adding it back again? Or having a `$this->is_currently_adding_token` flag to stop the execution of the function at the very start?

Reply:
I've seen "removing and adding the filter" as a common pattern for fixing these kinds of issues. It's a more direct fix for the problem, the flag seems like a workaround.

Reply:
> I've seen "removing and adding the filter" as a common pattern for fixing these kinds of issues.

Cool! Both solutions seemed a bit hacky, so at least I picked the preferred one by random chance :)